tests/test_integration_autodoc_type_aliases.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import re
+import sys
+from pathlib import Path
+from textwrap import dedent, indent
+from typing import TYPE_CHECKING, Any, Callable, Literal, NewType, TypeVar  # no type comments
+
+import pytest
+
+if TYPE_CHECKING:
+    from io import StringIO
+
+    from sphinx.testing.util import SphinxTestApp
+
+T = TypeVar("T")
+W = NewType("W", str)
+
+
+def expected(expected: str, **options: dict[str, Any]) -> Callable[[T], T]:
+    def dec(val: T) -> T:
+        val.EXPECTED = expected
+        val.OPTIONS = options
+        return val
+
+    return dec
+
+
+def warns(pattern: str) -> Callable[[T], T]:
+    def dec(val: T) -> T:
+        val.WARNING = pattern
+        return val
+
+    return dec
+
+
+ArrayLike = Literal["test"]
+
+
+@expected(
+    """\
+mod.function(x)
+
+   Function docstring.
+
+   Parameters:
+      **x** (ArrayLike) -- foo
+
+   Returns:
+      something
+
+   Return type:
+      bytes
+""",
+)
+def function(x: ArrayLike) -> str:  # noqa: ARG001
+    """
+    Function docstring.
+
+    :param x: foo
+    :return: something
+    :rtype: bytes
+    """
+
+
+AUTO_FUNCTION = ".. autofunction:: mod.{}"
+
+LT_PY310 = sys.version_info < (3, 10)
```

Comment:
Inline these two to where they are used.

Reply:
done